### PR TITLE
Add recipe for scikit-datasets.

### PR DESCRIPTION
**Pull request by vnmabus**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by vnmabus** [comment]
I confirm I am willing to be listed there.

**Comment by vnmabus** [comment]
@conda-forge/staged-recipes I think this is ready to review. I am unsure of what to do with the optional dependencies in `extras_require`, as Conda does not support optional dependencies, AFAIK.

File: recipes/scikit-datasets/scikit-datasets/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - pytest-runner
+    - python
+  run:
+    - numpy
+    - python
```
Comment (dopplershift) [inline recipes/scikit-datasets/scikit-datasets/meta.yaml:24]: Need to specify a version for Python:
```suggestion
    - python >=3.8
  run:
    - numpy
    - python >=3.8
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just needs a minor tweak.

For optional dependencies:
1. If you think it's really important and most users will want it, you can add it as a regular dependency
2.  User [multiple outputs](https://github.com/conda-forge/gstreamer-feedstock/blob/main/recipe/meta.yaml) from the recipe to create a second package that includes some/all of the optional ones
3. Just rely on users installing what they need when they need it.

**Comment by vnmabus** [comment]
> Just needs a minor tweak.
> 
> For optional dependencies:
> 
>     1. If you think it's really important and most users will want it, you can add it as a regular dependency
> 
>     2. User [multiple outputs](https://github.com/conda-forge/gstreamer-feedstock/blob/main/recipe/meta.yaml) from the recipe to create a second package that includes some/all of the optional ones
> 
>     3. Just rely on users installing what they need when they need it.

I will take option 3 for now, and study option 2 in the future.

**Review by dopplershift — CHANGES_REQUESTED** [review]
I didn't notice this before, but currently your recipe is located in `recipes/scikit-datasets/scikit-datasets/meta.yaml`. It should instead be located in `recipes/scikit-datasets/meta.yaml`.

**Comment by vnmabus** [comment]
> I didn't notice this before, but currently your recipe is located in `recipes/scikit-datasets/scikit-datasets/meta.yaml`. It should instead be located in `recipes/scikit-datasets/meta.yaml`.

Yup, my bad. Fixed now.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**